platform/parser/xml/src/main/java/org/codice/ddf/parser/xml/XmlParser.java

```diff
@@ -69,20 +76,89 @@ public JAXBContext load(CacheKey cacheKey) throws Exception {
 
     @Override
     public ParserConfigurator configureParser(List<String> contextPath, ClassLoader loader) {
-        return new XmlParserConfigurator().
-                setContextPath(contextPath).
-                setClassLoader(loader);
+        return new XmlParserConfigurator().setContextPath(contextPath)
+                .setClassLoader(loader);
     }
 
     @Override
     public void marshal(ParserConfigurator configurator, Object obj, OutputStream os)
             throws ParserException {
+        marshal(configurator, marshaller -> {
+            try {
+                marshaller.marshal(obj, os);
+            } catch (JAXBException e) {
+                throw new RuntimeException("Error marshalling", e);
+            }
+        });
+    }
+
+    @Override
+    public void marshal(ParserConfigurator configurator, Object obj, Node node)
+            throws ParserException {
+        marshal(configurator, marshaller -> {
+            try {
+                marshaller.marshal(obj, node);
+            } catch (JAXBException e) {
+                throw new RuntimeException("Error marshalling", e);
+            }
+        });
+    }
+
+    @Override
+    public <T> T unmarshal(ParserConfigurator configurator, Class<? extends T> cls,
+            final InputStream stream) throws ParserException {
+        return unmarshal(configurator, unmarshaller -> {
+            try {
+                XMLStreamReader xmlStreamReader = XMLInputFactory.newInstance()
+                        .createXMLStreamReader(stream);
+
+                @SuppressWarnings("unchecked")
+                T unmarshal = (T) unmarshaller.unmarshal(xmlStreamReader);
+                return unmarshal;
+            } catch (XMLStreamException | JAXBException e) {
+                throw new RuntimeException("Error unmarshalling", e);
+            }
+        });
+    }
+
+    @Override
+    public <T> T unmarshal(ParserConfigurator configurator, Class<? extends T> cls,
+            Node node) throws ParserException {
```

Comment:
Just noticed that the InputStream version of unmarshal declares the third parameter as "final", should the Node and Source versions do this as well?